hlint.cabal

```diff
@@ -13,17 +13,17 @@ description:
     HLint gives suggestions on how to improve your source code.
 homepage:           https://github.com/ndmitchell/hlint#readme
 bug-reports:        https://github.com/ndmitchell/hlint/issues
-data-dir:           data
 data-files:
-    hlint.yaml
-    default.yaml
-    Test.hs
-    report_template.html
-    hs-lint.el
-    hlint.1
-    hlint.ghci
-    HLint_QuickCheck.hs
-    HLint_TypeCheck.hs
+    data/hlint.yaml
+    data/default.yaml
+    data/Test.hs
+    data/report_template.html
+    data/hs-lint.el
+    data/hlint.1
+    data/hlint.ghci
+    data/HLint_QuickCheck.hs
+    data/HLint_TypeCheck.hs
+    src/Hint/*.hs
```

Comment:
Can you comment on why this change?

Reply:
The summary of the built-in hints is generated by running the test cases in `src/Hint/*.hs`. Adding these files to `data-files` ensures they are installed together with HLint.

Reply:
I'm no expert on these matters (!) but perhaps these `.hs` files should be in the `extra-source-files` stanza rather than the `data-files` stanza? If so then the `data-dir` property could maybe be restored?

Reply:
I'm not expert either, but `data-files` are available at runtime via `getDataFileName`, while `extra-source-files` don't seem to be. These files are needed when running `hlint summary`.

Reply:
See https://cabal.readthedocs.io/en/stable/developing-packages.html#pkg-field-extra-source-files. Not sure if it's got the properties we need of it. Just to be clear, I have no objection to the files being data, just trying to make sure that we're doing it the "right" way 😄

Reply:
It's great how documented cabal (and stack too..) is and how to do stuff using it. -_-